airflow/contrib/hooks/gcp_sheets_hook.py

```diff
@@ -0,0 +1,274 @@
+# specific language governing permissions and limitations
+# under the License.
+#
+"""
+This module contains a Google Sheets API hook
+"""
+
+from googleapiclient.discovery import build
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
+from airflow.exceptions import AirflowException
+
+
+class GCPSheetsHook(GoogleCloudBaseHook):
+    """
+    Interact with Google Sheets via GCP connection
+    Reading and writing cells in Google Sheet:
+    https://developers.google.com/sheets/api/guides/values
+    """
+
+    def __init__(self,
+                 gcp_conn_id: str = 'google_cloud_default',
+                 spreadsheet_id: str = None,
+                 api_version: str = 'v4',
+                 delegate_to: str = None) -> None:
+        """
+        :param grpc_conn_id: The connection ID to use when fetching connection info.
+        :type grpc_conn_id: str
+        :param gsheet_id: The Google Sheet ID to interact with
+        :type gsheet_id: str
+        :param range_: The Range or Named Range to interact with
+        :type range_: str
+        """
+        super().__init__(gcp_conn_id, delegate_to)
+        self.gcp_conn_id = gcp_conn_id
+
+        if not spreadsheet_id:
+            raise AirflowException("The spreadsheet_id must be passed!")
+
+        self.spreadsheet_id = spreadsheet_id
+        self.api_version = api_version
+        self.delegate_to = delegate_to
+        self.num_retries = self._get_field('num_retries', 1)
+        self._conn = None
+
+    def get_conn(self):
+        """
+        Retrieves connection to Google Sheets.
+        :return: Google Sheets services object.
+        :rtype: dict
+        """
+        if not self._conn:
+            http_authorized = self._authorize()
+            self._conn = build('sheets', self.api_version, http=http_authorized, cache_discovery=False)
+
+        return self._conn
+
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    def get_values(self,
+                   range_: str,
+                   major_dimension: str = 'DIMENSION_UNSPECIFIED',
+                   value_render_option: str = 'FORMATTED_VALUE',
+                   date_time_render_option: str = 'SERIAL_NUMBER',
+                   project_id: str = None) -> dict:
+        """
+        Gets values from Google Sheet from a single range
+        https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets.values/get
+        """
+        assert project_id is not None
+        if not range_:
+            raise AirflowException("The 'range_' argument must be passed!")
+        service = self.get_conn()
+        response = service.spreadsheets().values().batchGet(  # pylint: disable=no-member
+            spreadsheetId=self.spreadsheet_id,
+            ranges=range_,
+            majorDimension=major_dimension,
+            valueRenderOption=value_render_option,
+            dateTimeRenderOption=date_time_render_option
+        ).execute()
```

Comment:
```suggestion
        ).execute(num_retries=self.num_retries)
```
If we have `self.num_retries` it is a good idea to use it in requests.